services/QuillLMS/app/services/analytics/cache_segment_school_data.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+class CacheSegmentSchoolData
+  CACHE_KEYS = [
+    TOTAL_TEACHERS_AT_SCHOOL = 'TOTAL_TEACHERS_AT_SCHOOL',
+    TOTAL_STUDENTS_AT_SCHOOL = 'TOTAL_STUDENTS_AT_SCHOOL',
+    TOTAL_ACTIVITIES_COMPLETED_BY_STUDENTS_AT_SCHOOL = 'TOTAL_ACTIVITIES_COMPLETED_BY_STUDENTS_AT_SCHOOL',
+    ACTIVE_TEACHERS_AT_SCHOOL_THIS_YEAR = 'ACTIVE_TEACHERS_AT_SCHOOL_THIS_YEAR',
+    ACTIVE_STUDENTS_AT_SCHOOL_THIS_YEAR = 'ACTIVE_STUDENTS_AT_SCHOOL_THIS_YEAR',
+    TOTAL_ACTIVITIES_COMPLETED_BY_STUDENTS_AT_SCHOOL_THIS_YEAR = 'TOTAL_ACTIVITIES_COMPLETED_BY_STUDENTS_AT_SCHOOL_THIS_YEAR'
+  ]
+
+  CACHE_LIFE = 60*60*25
+
+  def initialize(school)
+    @school = school
+  end
+
+  def read(namespace)
```

Comment:
+1 for adding a namespace, but IMO it should be a constant in this class, to avoid other values getting passed into the `read` and `write` methods. Thoughts?